CMakeModules/FindQHULL.cmake

```diff
@@ -1,42 +1,30 @@
-# Find qhull header and library.
+# Find QHULL
 #
+# This sets the following variables:
+# QHULL_FOUND - True if QHULL was found.
+# QHULL_INCLUDE_DIRS - Directories containing the QHULL include files.
+# QHULL_LIBRARIES - Libraries needed to use QHULL.
 
-# This module defines the following uncached variables:
-#  QHULL_FOUND, if false, do not try to use qhull.
-#  QHULL_INCLUDE_DIRS, where to find qhull/qhull_a.h.
-#  QHULL_LIBRARIES, the libraries to link against to use the qhull library
-#  QHULL_LIBRARY_DIRS, the directory where the qhull library is found.
+find_path(QHULL_INCLUDE_DIR NAMES libqhullcpp/Qhull.h
+        HINTS ${CMAKE_PREFIX_PATH}/src /usr/local/include /usr/include
+        PATH_SUFFIXES qhull src/libqhull libqhull include
+        )
 
-find_path(
-  QHULL_INCLUDE_DIR
-  qhull/qhull_a.h libqhull/qhull_a.h
-  PATHS /usr/local/include /usr/include
-)
+if(QHULL_INCLUDE_DIR)
+    find_library(QHULL_R_LIBRARY NAMES qhull_r HINTS ${CMAKE_PREFIX_PATH})
+    find_library(QHULL_CPP_LIBRARY NAMES qhullcpp HINTS ${CMAKE_PREFIX_PATH})
```

Comment:
On my machine, there is no `libqhullcpp` library but only `libqhull`.

Reply:
yes, @eloirozier mentioned that this file appeared in version 8.0+ which is not available on most Linux Distros
I got it doing : 
```
mkdir Qhull_Git
cd Qhull_Git/
git clone https://github.com/qhull/qhull
cd qhull/
make
[...]
ls lib/
libqhullcpp.a  libqhull_r.so  libqhull_r.so.8.0  libqhull_r.so.8.0.2  libqhullstatic.a  libqhullstatic_r.a
```

Reply:
how about:
```
    find_library(QHULL_CPP_LIBRARY NAMES qhullcpp qhull HINTS ${CMAKE_PREFIX_PATH})
```

Reply:
I did some research.
1. On Ubuntu Focal, I have tried downloading and compiling `https://github.com/qhull/qhull/releases/tag/2015.2`. This generates `libqhullcpp.a`. When I compiled this PR with the resulting file, it worked fine.
2. On Fedora 34, the QHull package is still 7.2.1 (qhull-devel-7.2.1-7), this also contains `libqhullcpp.a`. I confirmed this PR compiles well.

Therefore I think only the QHull library 7 also provides the `libqhullcpp.a`, but somehow it is not included in the Ubuntu Focal package. Since this is a static library, perhaps we can ship the binary for Ubuntu Focal.

Also some more things:

1. On SuSE Tumbleweed, you will need 2 packages: `qhull_r-devel` and `qhullcpp-devel-static`. After installing the packages, this PR compiles well.
2. On Windows, although the QHull package available via MSYS2 is the latest (2020.2), this results in a link error at the end, this is weird.
3. On macOS, CMake fails to find the libraries, although the one provided by Homebrew is the latest (2020.2). This is weird too.